File: Assets/Scripts/Inventory/Currency/Currency.cs
using UnityEngine;
[System.Serializable]
public class Currency
{
    [SerializeField]
    private int _amount;
    [SerializeField]
    private string _name;
    public int Amount { get => _amount; set => _amount = value; }
    public string Name { get => _name; set => _name = value; }
}

[System.Serializable]
public class Gold : Currency
{
    public Gold(int amount)
    {
        Name = "Gold";
        Amount = amount;
    }
}

File: Spring/siko/src/main/kotlin/com/siko25/siko/document/Currency.kt
package com.siko25.siko.document

import org.springframework.data.annotation.Id
import org.springframework.data.mongodb.core.mapping.Document

@Document
data class Currency(
        @Id override val id: String = "0",
        val amount: Int,
        override val name: String = "Gold",
        override val type: List<String> = listOf("Currency"),
        val rarity: String = "Common",
) : Item
